core-bundle/src/DataCollector/ContaoDataCollector.php

```diff
@@ -148,6 +168,44 @@ private function addSummaryData(): void
             'preview' => \defined('BE_USER_LOGGED_IN') && true === BE_USER_LOGGED_IN,
             'layout' => $this->getLayoutName(),
             'template' => $this->getTemplateName(),
+            'legacy_routing' => $this->parameterBag->get('contao.legacy_routing'),
+        ];
+    }
+
+    private function addLegacyRoutingData(): void
+    {
+        $hooks = [];
+
+        foreach (['getPageIdFromUrl', 'getRootPageFromUrl'] as $name) {
+            if (empty($GLOBALS['TL_HOOKS'][$name]) || !\is_array($GLOBALS['TL_HOOKS'][$name])) {
+                continue;
+            }
+
+            /** @var System $systemAdapter */
+            $systemAdapter = $this->framework->getAdapter(System::class);
+
+            foreach ($GLOBALS['TL_HOOKS'][$name] as $callback) {
+                $class = $systemAdapter->importStatic($callback[0]);
+                $r = new \ReflectionClass($class);
+                $file = $r->getFileName();
+                $vendorDir = $this->parameterBag->get('kernel.project_dir').'/vendor/';
```

Comment:
I'm not sure that's correct, I explicitly want the slash suffix? Also, because the value is only passed to the `Path` method, its canonicalized anyway?

Reply:
I don't understand why you need the `/` suffix? That shouldn't make any difference for the explode?

Reply:
well what if the other path is '.../vendordir/`? Like the same name but longer?

Reply:
In `[$vendor, $package] = explode('/', '$path, 3)` vendor and package will always be the two parts around the first slash, no matter if there is another one (or other parts) coming afterwards. So it'd be the same for a paths `a/b/c/`, `a/b/`, `a/b`.

Reply:
+1 for using Path::join().

Reply:
> In `[$vendor, $package] = explode('/', '$path, 3)` vendor and package will always be the two parts around the first slash, no matter if there is another one (or other parts) coming afterwards. So it'd be the same for paths `a/b/c/`, `a/b/`, `a/b`.

That's not what I mean. If `$file` is `/some/path/vendordir/file.php` and `$vendorDir` is `/some/path/vendor`, it would match but shouldn't, but `/some/path/vendor/` does correctly not match.

Reply:
But then `Path::isBasePath` would return false?

Reply:
@aschempp You should probably add an explaining comment so we still know in 6 months. 😄